pandas/_libs/lib.pyx

```diff
@@ -2247,6 +2258,28 @@ def maybe_convert_numeric(
     cdef:
         object val = values[0]
 
+    # Convert python strings into ones readable by C
+
+    cdef char* tsep
+    cdef char* dsep
+    # Use null char to represent lack of separator
+    if thousands is None:
+        tsep = "\0"
+    else:
+        bytes_tsep = thousands.encode("UTF-8")
```

Comment:
Doesn't `thousands.encode` return a `PyObject *`? Assigning that to `tsep` does not seem correct

Reply:
Cython handles the conversion when assigning bytes_tsep to tsep and I pretty much just copied the code from the documentation: https://cython.readthedocs.io/en/latest/src/tutorial/strings.html#encoding-text-to-bytes
If there is a way solve the char dtype issue as mentioned above then this can be simplified though.